OpenRA.Mods.RA2/Traits/Render/WithSupportPowerChargedOverlay.cs

```diff
@@ -0,0 +1,85 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.Common.Traits.Render;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits.Render
+{
+	[Desc("Displays an overlay when `SupportPower` is fully charged.")]
+	public class WithSupportPowerChargedOverlayInfo : ConditionalTraitInfo, Requires<RenderSpritesInfo>, Requires<BodyOrientationInfo>
```

Comment:
This should also have `Requires<SupportPowerInfo>`.

Reply:
No, it is loosely coupled with interfaces.

Reply:
It won't work at all on an actor that doesn't have a `SupportPower` trait.

Reply:
See [WithSupportPowerActivationOverlay](https://github.com/OpenRA/OpenRA/blob/a261abbc3f6c5dbf80b0e5132a0071efdd099bc5/OpenRA.Mods.Common/Traits/Render/WithSupportPowerActivationOverlay.cs#L18). This is only to be used for trait queries that otherwise crash the game. The first lines already mentions everything you need to know for the documentation.